packages/keystone/tasks.js

```diff
@@ -38,23 +33,65 @@ const CRON_TASKS = new Map()
 const REMOVE_CRON_TASKS = []
 let isWorkerCreated = false
 
-function createTask (name, fn, opts = {}) {
+/**
+ * Create bull queue. For internal use only!
+ * @param name {string} name of bull queue
+ */
+function createTaskQueue (name) {
+    if (IS_BUILD) return
+
+    QUEUES.set(name, new Queue(name, {
+        /**
+         * @param {'client' | 'subscriber' | 'bclient'} type
+         * @return {import('ioredis')}
+         */
+        createClient: (type, opts) => {
+            // NOTE(pahaz): https://github.com/OptimalBits/bull/issues/1873
+            //   Bull uses three Redis connection. Probably, we can share regular Redis connection for type 'client' think about it!
+            if (['bclient', 'subscriber'].includes(type)) {
+                opts.maxRetriesPerRequest = null
+            }
+            return getRedisClient(`worker:${name}`, type, opts)
+        },
+    }))
+}
+
+/**
+ * Creates and register task. After that you can call it with delay or applyAsync methods
+ * @param name {String} name of the task
+ * @param fn {Function} function that will be executed at the worker side
+ * @param queue {String} task queue to which this function will be added. Can be used for change priority of the task execution
+ * @param opts {Object} bull task options
+ * @return {(function(): never)|*}
+ */
+function createTask (name, fn, queue = DEFAULT_QUEUE_NAME, opts = {}) {
     if (typeof fn !== 'function') throw new Error('unsupported fn argument type. Function expected')
     if (!name) throw new Error('no name')
+    if (get(opts, 'priority')) throw new Error('Deprecated task parameter priority. Please don\'t use it. You can change task running priority via queue property')
 
     if (TASKS.has(name)) throw new Error(`Task with name ${name} is already registered`)
     TASKS.set(name, fn)
-    return createTaskWrapper(name, fn, opts)
+    return createTaskWrapper(name, fn, queue, opts)
 }
 
-function createCronTask (name, cron, fn, opts = {}) {
+/**
+ * Creates and register cron task. It will be called at the defined period of time
+ * @param name {String} name of cron task
+ * @param cron {String} cron style definition string. For example, 0 12 * * *
+ * @param fn {Function} function that will be executed at the worker side
+ * @param queue {String} task queue to which this function will be added. Can be used for change priority of the task execution
+ * @param opts {Object} bull task options
+ * @return {(function(): never)|*}
+ */
+function createCronTask (name, cron, fn, queue = DEFAULT_QUEUE_NAME, opts = {}) {
     if (typeof fn !== 'function') throw new Error('unsupported fn argument type. Function expected')
```

Comment:
Sam about queue name validation

Reply:
What kind of validation did you mean? There is actually two ways to register bull queues. And we picked up the way, that allow you to define queue at prepareKeystone stage.
That point means, that task creation will triggered much earlier than queues creation. So you can't check anything with this approach at the current context.